tests/serverpod_test_server/lib/src/endpoints/citizen_endpoint.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:serverpod/serverpod.dart';
+
+class CitizenEndpoint extends Endpoint {
+  Future<List<Citizen>> getAllShallow(Session session) async {
+    return await Citizen.find(session, orderBy: Citizen.t.id);
+  }
+
+  Future<List<Citizen>> getAllWithDeepIncludes(Session session) async {
+    return await Citizen.find(
+      session,
+      orderBy: Citizen.t.id,
+      include: CitizenInclude(
+        company: CompanyInclude(town: TownInclude()),
+        oldCompany: CompanyInclude(town: TownInclude()),
+      ),
+    );
+  }
+
+  Future<int> insert(Session session, Citizen citizen) async {
+    await Citizen.insert(session, citizen);
+    return citizen.id!;
```

Comment:
Question: How do we want to deal with data creation / updates ? We can perhaps add this in a different PR, but I think this part is necessary to be feature complete with the relation feature. We can discuss this offline.

Reply:
I think this is a separate discussion and possibly separate feature. Let's discuss it offline.